akka-stream-tests/src/test/scala/akka/stream/scaladsl/UnfoldResourceAsyncSourceSpec.scala

```diff
@@ -0,0 +1,246 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.scaladsl
+
+import java.io._
+
+import akka.Done
+import akka.actor.{ NoSerializationVerificationNeeded, ActorSystem }
+import akka.stream.ActorAttributes._
+import akka.stream.Supervision._
+import akka.stream.{ ActorMaterializer, _ }
+import akka.stream.impl.StreamSupervisor.Children
+import akka.stream.impl.{ ActorMaterializerImpl, StreamSupervisor }
+import akka.stream.testkit.TestSubscriber
+import akka.stream.testkit.Utils._
+import akka.stream.testkit.scaladsl.TestSink
+import akka.util.{ ByteString, Timeout }
+import akka.testkit.AkkaSpec
+
+import scala.concurrent.{ Future, Promise }
+import scala.concurrent.duration._
+import scala.util.control.NoStackTrace
+
+class UnfoldResourceAsyncSourceSpec extends AkkaSpec(UnboundedMailboxConfig) {
+
+  val settings = ActorMaterializerSettings(system).withDispatcher("akka.actor.default-dispatcher")
+  implicit val materializer = ActorMaterializer(settings)
+
+  val manyLines = {
+    ("a" * 100 + "\n") * 10 +
+      ("b" * 100 + "\n") * 10 +
+      ("c" * 100 + "\n") * 10 +
+      ("d" * 100 + "\n") * 10 +
+      ("e" * 100 + "\n") * 10 +
+      ("f" * 100 + "\n") * 10
+  }
+  val manyLinesArray = manyLines.split("\n")
+
+  val manyLinesFile = {
+    val f = File.createTempFile("blocking-source-async-spec", ".tmp")
+    new FileWriter(f).append(manyLines).close()
+    f
+  }
```

Comment:
This can be any resource like connection or stream. Or anything that we have to create/read/destroy or open/read/close. So I believe it's mainly still about IO.